app/controllers/session_controller.rb

```diff
@@ -172,32 +172,47 @@ def sso_login
     sso.expire_nonce!
 
     begin
+      validate_invitiation!(sso)
```

Comment:
Instead of setting the @invite instance variable in this method, I find the code much clearer to read if this method returns an `invite` object.